pychromecast/dial.py

```diff
@@ -101,9 +100,24 @@ def _get_status(
     if secure and not has_context:
         context = get_ssl_context()
 
-    req = urllib.request.Request(url, headers=headers)
-    with urllib.request.urlopen(req, timeout=timeout, context=context) as response:
-        data = response.read()
+    try:
+        req = urllib.request.Request(url, headers=headers)
+        with urllib.request.urlopen(req, timeout=timeout, context=context) as response:
+            data = response.read()
+    except urllib.error.HTTPError as err:
+        if err.code != HTTPStatus.FORBIDDEN:
+            raise
+        # We may be blocked because we're connecting to a hostname not matching the hostname
+        # published by the chromecast device via mDNS. Do another attempt with an empty
+        # host header.
```

Comment:
Maybe I'm lying here; I'm assuming the device will accept the hostname it itself publishes. Or do devices only accept no host header or host header set to the stringified IP?

Reply:
I tested on one of my own devices, and it doesn't accept the HTTP request also when `host` is set to the hostname it itself publishes.